src/systemtraymanager.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright (C) 2012 Sebastian Herbord. All rights reserved.
+
+This file is part of Mod Organizer.
+
+Mod Organizer is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Mod Organizer is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Mod Organizer.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "systemtraymanager.h"
+
+#include <QAction>
+#include <QDockWidget>
+#include <QIcon>
+#include <QMainWindow>
+#include <QMenu>
+#include <QSystemTrayIcon>
+
+SystemTrayManager::SystemTrayManager(QMainWindow* parent, QDockWidget* logDock)
+    : m_Parent(parent), m_LogDock(logDock), m_SystemTrayIcon(nullptr)
+{}
+
+SystemTrayManager::~SystemTrayManager()
+{
+  if (m_SystemTrayIcon) {
+    delete m_SystemTrayIcon;
+  }
+}
+
+void SystemTrayManager::minimizeToSystemTray()
+{
+  if (!m_SystemTrayIcon) {
+    createSystemTrayIcon();
```

Comment:
Is there any reason not to initialize `m_SystemTrayIcon` in the constructor? This would avoid a lot of check everywhere, and would remove the need for a custom destructor since Qt's ownership would delete the object for us.